app/jinja_filters.py

```diff
@@ -142,6 +152,42 @@ def format_conditional_date(context, date1=None, date2=None):
     return format_date(context, date)
 
 
+@blueprint.app_template_filter()
+def calculate_offset_date(input_datetime, offset_days, day_of_week='SU'):
+    """
+    Offset a date from a particular day of the week in the previous week.
+
+    Intended to be used to generate reference date ranges around a particular date
+    The offsets will always be based on one day of the week in the previous Mon-Sun
+
+    :param (str) input_datetime: The datetime (or date) to offset. Defaults to now
+    :param (int) offset_days: The number of days to offset this date`
+    :param (str) day_of_week: The day of the previous week to offset from (two letter abbreviation)
+    :returns (str): The offset date
+    """
+    if input_datetime:
+        if 'T' in input_datetime:
+            parsed_datetime = datetime.strptime(input_datetime, '%Y-%m-%dT%H:%M:%S.%f')
+        else:
+            parsed_datetime = datetime.strptime(input_datetime, '%Y-%m-%d')
+    else:
```

Comment:
Not sure I like this. Possibly use result of `input_datetime.split("T")[0]` instead.
```
if input_datetime:
   parsed_datetime = datetime.strptime(input_datetime.split("T")[0], '%Y-%m-%d')
else:
```